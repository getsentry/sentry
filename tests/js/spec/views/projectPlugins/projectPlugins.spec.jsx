import React from 'react';

import {mountWithTheme} from 'sentry-test/enzyme';

import ProjectPlugins from 'app/views/settings/projectPlugins/projectPlugins';

describe('ProjectPlugins', function() {
  let wrapper;
  const routerContext = TestStubs.routerContext();
  const plugins = TestStubs.Plugins();
  const org = TestStubs.Organization();
  const project = TestStubs.Project();
  const params = {
    orgId: org.slug,
    projectId: project.slug,
  };

  it('renders', function() {
    wrapper = mountWithTheme(
      <ProjectPlugins params={params} plugins={plugins} />,
      routerContext
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('has loading state', function() {
    wrapper = mountWithTheme(
      <ProjectPlugins params={params} loading plugins={[]} />,
      routerContext
    );

    expect(wrapper.find('LoadingIndicator')).toHaveLength(1);
  });

  it('has error state when plugins=null and loading is true', function() {
    wrapper = mountWithTheme(
      <ProjectPlugins
        params={params}
        plugins={null}
        loading
        error={new Error('An error')}
      />,
      routerContext
    );

    expect(wrapper.find('RouteError')).toHaveLength(1);
  });

  it('has error state when plugins=[]', function() {
    wrapper = mountWithTheme(
      <ProjectPlugins
        params={params}
        plugins={[]}
        loading
        error={new Error('An error')}
      />,
      routerContext
    );
    expect(wrapper.find('RouteError')).toHaveLength(1);
  });
});
