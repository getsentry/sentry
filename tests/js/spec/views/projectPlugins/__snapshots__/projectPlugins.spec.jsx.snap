// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPlugins renders 1`] = `
<Panel>
  <PanelHeader>
    <div>
      Legacy Integration
    </div>
    <div>
      Enabled
    </div>
  </PanelHeader>
  <PanelBody
    direction="column"
    disablePadding={true}
    flex={false}
  >
    <PanelAlert
      m={0}
      mb={0}
      type="warning"
    >
      Legacy Integrations must be configured per-project. It's recommended to prefer organization-integrations over the legacy project integrations when available, as features will be more robust and flexible.
    </PanelAlert>
    <PanelItem
      key="amazon-sqs"
      p={2}
    >
      <ProjectPluginRow
        assets={Array []}
        author={
          Object {
            "name": "Sentry Team",
            "url": "https://github.com/getsentry/sentry",
          }
        }
        canDisable={true}
        enabled={false}
        hasConfiguration={true}
        id="amazon-sqs"
        name="Amazon SQS"
        params={
          Object {
            "orgId": "org-slug",
            "projectId": "project-slug",
          }
        }
        slug="amazon-sqs"
        version="8.23.0.dev0"
      />
    </PanelItem>
    <PanelItem
      key="github"
      p={2}
    >
      <ProjectPluginRow
        assets={Array []}
        author={
          Object {
            "name": "Sentry Team",
            "url": "https://github.com/getsentry/sentry",
          }
        }
        canDisable={false}
        enabled={true}
        id="github"
        name="GitHub"
        params={
          Object {
            "orgId": "org-slug",
            "projectId": "project-slug",
          }
        }
        slug="github"
        version="8.23.0.dev0"
      />
    </PanelItem>
  </PanelBody>
</Panel>
`;
